### PR TITLE
Update response chapter.

### PR DESCRIPTION
**Pull request by markstory**
* Document new interfaces.
* Include section on common mistakes with immutable objects.

Refs cakephp/cakephp#9636

### CODE REVIEW

File: en/controllers/request-response.rst [hunk 1]
```diff
@@ -829,21 +824,32 @@ related headers with a fluent interface::
...
-2. The request's ``Origin`` value matches one of the allowed Origin values.
+#. The request has an ``Origin`` header.
+#. The request's ``Origin`` value matches one of the allowed Origin values.
 
 .. versionadded:: 3.2
     The ``CorsBuilder`` was added in 3.2
 
-Sending the Response
---------------------
+Common Mistakes with Immutable Responses
```
Comment (ravage84) [inline en/controllers/request-response.rst]: I feel this note is way too far down the page.

File: en/controllers/request-response.rst [hunk 2]
```diff
@@ -829,21 +824,32 @@ related headers with a fluent interface::
...
 
-Once you are done creating a response, calling ``send()`` will send all the set
-headers as well as the body. This is done automatically at the end of each
-request by ``Dispatcher``.
+    $this->response = $this->response->withHeader('X-CakePHP', 'yes!');
 
 .. meta::
     :title lang=en: Request and Response objects
-    :keywords lang=en: request controller,request parameters,array indexes,purpose index,response objects,domain information,request object,request data,interrogating,params,previous versions,introspection,dispatcher,rout,data structures,arrays,ip address,migration,indexes,cakephp
+    :keywords lang=en: request controller,request parameters,array indexes,purpose index,response objects,domain information,request object,request data,interrogating,params,previous versions,introspection,dispatcher,rout,data structures,arrays,ip address,migration,indexes,cakephp,PSR-7,immutable
```
Comment (ravage84) [inline en/controllers/request-response.rst]: PSR-7 in uppercase?

File: en/controllers/request-response.rst [hunk 3]
```diff
@@ -68,12 +68,14 @@ Query String Parameters
 
 .. php:method:: query($name)
 
-Query string parameters can be read using
-:php:attr:`~Cake\\Network\\Request::$query`::
+Query string parameters can be read using the ``getQuery()`` method::
 
     // URL is /posts/index?page=1&sort=title
     $this->request->getQuery('page');
```
Comment (ravage84) [inline en/controllers/request-response.rst]: I would write ``$page = $this->request->getQuery('page');``, this way it becomes even more clear. I think this is nov more important than before, as we now prefix all setter methods with a ``$response = ``.

Shall I open a separate PR after this gets merged?

File: en/controllers/request-response.rst (hunk 2 above)
Comment (markstory) [inline en/controllers/request-response.rst]: Sure!

File: en/controllers/request-response.rst (hunk 1 above)
Comment (markstory) [inline en/controllers/request-response.rst]: I thought it was kind of awkward at other positions in the page.

File: en/controllers/request-response.rst (hunk 3 above)
Comment (markstory) [inline en/controllers/request-response.rst]: sure!

Comment (markstory) [inline en/controllers/request-response.rst]: Thanks!

**Pull request merged**